src/ApiService/ApiService/Functions/TimerTasks.cs

```diff
@@ -22,35 +23,39 @@ public class TimerTasks {
 
     [Function("TimerTasks")]
     public async Async.Task Run([TimerTrigger("00:00:15")] TimerInfo myTimer) {
-        var expriredTasks = _taskOperations.SearchExpired();
-        await foreach (var task in expriredTasks) {
+        // perform up to 10 updates in parallel for each entity type
+        var parallelOptions = new ParallelOptions { MaxDegreeOfParallelism = 10 };
+
+        var expiredTasks = _taskOperations.SearchExpired();
+        // marking one task stopping can also mark other tasks in the same job as failed
+        // this could make parallel updates stomp on each other, so don't do it in parallel
+        await foreach (var task in expiredTasks) {
             _logger.LogInformation("stopping expired task. job_id:{JobId} task_id:{TaskId}", task.JobId, task.TaskId);
             await _taskOperations.MarkStopping(task, "task is expired");
         }
 
-
         var expiredJobs = _jobOperations.SearchExpired();
-
-        await foreach (var job in expiredJobs) {
+        // job updates are all distinct and only update tasks owned by that job, can be performed in parallel
+        await Parallel.ForEachAsync(expiredJobs, parallelOptions, async (job, _cancel) => {
```

Comment:
Another option here is to have an overload to Stopping that takes a list of jobs. We can update all of them in batch we would also save in the number of connection to the db.

Reply:
You can only do batch updates within a single partition. Unfortunately, for the jobs each job lives in a separate partition so we cannot do this.